packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -120,7 +120,11 @@ export interface ClusterAttributes {
  */
 export interface ClusterOptions {
   /**
-   * The VPC in which to create the Cluster
+   * The VPC in which to create the Cluster.
+   *
+   * Note that if `endpointAccess` is configured to private only, the VPC must
```

Comment:
I expect these options to automatically be configured on the vpc.

Reply:
All these options are configured by default when creating a VPC. This means that if we create a VPC on the user's behalf, everything is ok.

If a VPC is given, there are two options:

1. An imported VPC - In this case we can't set these options, nor can we validate them (right?)
2. A new VPC - So again, by default these options are configured, the only way they aren't is if a user explicitly passed them to the VPC:

```typescript
new ec2.Vpc(this, 'Vpc', {enableDnsHostnames: false, enableDnsSupport: false})
```

In which case, I don't think we should override the configuration. My plan was to simply add a validation in case the vpc is a `Vpc` (and not an `IVpc`). This would require to either expose them in the L2, or use `vpc.node.defaultChild` and inspect the L1 resource. 

WDYT?

Reply:
Ideally we should be able to query the `Vpc` construct (if it is `instanceof ec2.Vpc`) to make sure they are enabled and error if they aren't so users can fix. If it's not too much of a hassle to add these readonly getters, let's do that. Otherwise, I am fine with the current situation.

Reply:
👍